src/codegen.cpp

```diff
@@ -6804,7 +6806,7 @@ static Function* gen_cfun_wrapper(
 
     jl_codectx_t ctx(M->getContext(), params);
     ctx.f = cw;
-    ctx.world = world;
+    ctx.min_world = ctx.max_world = world;
```

Comment:
It seems a little bit tricky to have `world` in both params and `ctx`, since it leads to odd cases like this. It might be easier to only have them in `ctx`?

Reply:
You mean only have them in the params, since the `ctx` has a reference to that?